lib/convert.ml

```diff
@@ -1,33 +1,156 @@
 open Prelude
 
+module type ATTACHMENT_CONVERTER =
+sig
+  val convert: string -> (string -> string)
+end
+
+module Attach_conv: ATTACHMENT_CONVERTER = struct
+  let convert script str =
+    let args = split script in
+      match Unix.Proc.rw args str with
+      | exception (Failure msg) ->
+          write stderr ("Conversion Failure: " ^ msg); str (* TODO: Better logging *)
+      | exception e -> raise e
+      | converted -> converted
+end
+
 module type CONVERT =
 sig
   type error
   type parsetree
-  type htransform = string -> string
-  type btransform = string -> string
   val parse : string -> (parsetree, error) result
+  val to_string : parsetree -> string
   val amap : Configuration.Formats.t -> parsetree -> (parsetree, error) result
   val acopy : Configuration.Formats.t -> parsetree -> (parsetree, error) result
-  val to_string : parsetree -> string
-  val convert : string -> (string -> string)
-  val acopy_email : (Configuration.Formats.t) -> string -> (string, error) result
+  val acopy_email : Configuration.Formats.t -> string -> (string, error) result
+  val acopy_mbox : Configuration.Formats.t -> in_channel -> (unit, error) result
 end
 
-module Conversion_ocamlnet = struct
+module Conversion_ocamlnet_F (C: ATTACHMENT_CONVERTER) = struct
```

Comment:
Nice.  One pattern OCamlers sometimes use for this type of thing is:
```ocaml
module CoolThing = struct
  module Make (M : COOLSIGNATURE) = struct
    (* ... *)
  end
end
```
Then when you wanna pass something into the module functor:
```
# module UseTheFunctor = CoolThing.Make (AwesomeInputModule);;
```